### PR TITLE
[Vue 3] Feat: add torrents modal

### PR DESCRIPTION
**Pull request by k4lu-0p**
Allows you to add one or more torrents to the queue via the AddModal.vue component

### CODE REVIEW

File: src/Pages/Dashboard.vue [hunk 1]
```diff
@@ -15,21 +15,15 @@ definePage({
 const mainData = useMainData()
 
 const torrents = computed(() => {
-  const torrents: Torrent[] = []
+  // temp
+  const _torrents = Object.values((mainData.data.value?.torrents as any))
```
Comment (WDaan) [inline src/Pages/Dashboard.vue:19]: Cool, but we should try to keep the typings as much as possible

File: src/composables/api/torrents.ts [hunk 2]
```diff
@@ -1,19 +1,115 @@
...
+
+export const useTorrents = () => {
+
+  async function addTorrents(payload: AddTorrentsPayload): Promise<any> {
+    
+    let requestBody: FormData | URLSearchParams | null = null;
+    const files: File[] = payload.files
+
+    // Map params
+    const params: QbAddTorrentPayload = {
```
Comment (WDaan) [inline src/composables/api/torrents.ts:59]: This can probably be done a bit more efficient

**Review by WDaan — COMMENTED** [review]
Sorry this took a while, public holidays and bussiness trip...

Awesome! thanks for picking this up. Can't wait to try this out.
We're still quite far away from actually moving over and releasing but at least we're making some progress.

File: src/composables/api/torrents.ts [hunk 3]
```diff
@@ -1,19 +1,115 @@
...
+      autoTMM: payload.autoTMM,
+      category: payload.category?.name,
+      cookie: payload.category?.name,
```
[the remark this answers is not in the document]
Comment (WDaan) [reply src/composables/api/torrents.ts:62]: Thanks @Larsluph for taking a look at this as well

File: src/components/Modals/AddModal.vue [hunk 4]
```diff
@@ -1,62 +1,354 @@
...
+const params = reactive<AddTorrentsPayload>({
+  urls: '',
+  files: [],
+  tags: [],
+  category: undefined,
+  directory: DEFAULT_DEST_FOLDER_PATH,
+  start: false,
+  autoTMM: false,
+  contentLayout: AppPreferences.ContentLayout.ORIGINAL,
+  stopCondition: AppPreferences.StopCondition.NONE,
```
Comment (Larsluph) [inline src/components/Modals/AddModal.vue:51]: Default value for these 3 is in settings

File: src/components/Modals/AddModal.vue [hunk 5]
```diff
@@ -1,62 +1,354 @@
...
+
+const isFormValid = ref(false)
+const isDropAreaVisible = ref(false)
+
+const params = reactive<AddTorrentsPayload>({
+  urls: '',
+  files: [],
+  tags: [],
+  category: undefined,
+  directory: DEFAULT_DEST_FOLDER_PATH,
```
Comment (Larsluph) [inline src/components/Modals/AddModal.vue:47]: for this one, it would be awesome to update the value when the category is changed

Comment (Larsluph) [reply src/components/Modals/AddModal.vue:47]: mb, haven't seen the watch ^^

File: src/composables/api/torrents.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (k4lu-0p) [reply src/composables/api/torrents.ts:62]: Oops! Thanks @Larsluph

**Comment by WDaan** [comment]
Why closing? 😛 I'm sure we can use most of it?

**Comment by k4lu-0p** [comment]
Sorry, issue with Git in my code editor, i'm fixing it

File: src/composables/api/torrents.ts (hunk 2 above)
Comment (k4lu-0p) [reply src/composables/api/torrents.ts:59]: Qbit payload properties do not have a constant format. There's camelCase, snakeCase and so on... I can't see any other way of solving this problem. Do you have a suggestion?

File: src/components/Modals/AddModal.vue (hunk 4 above)
Comment (k4lu-0p) [reply src/components/Modals/AddModal.vue:51]: settings/preferences are loaded when the modal has been opened

**Pull request closed without merging**